### PR TITLE
Improve error-handling from collectors (OSV only thus far)

### PR DESCRIPTION
**Pull request by bobmcwhirter**
through collectorist and up to exhort and the analyze request.

### CODE REVIEW

File: collector/osv/Cargo.toml [hunk 1]
```diff
@@ -28,6 +28,7 @@ serde_json = "1.0.68"
 tokio = { version = "1.0", features = ["full"] }
 guac = { workspace = true }
 packageurl = "0.3.0"
+url = "*"
```
Comment (jcrossley3) [inline collector/osv/Cargo.toml:31]: This smells funny. Intentional?

File: collectorist/api/src/server/collect.rs [hunk 2]
```diff
@@ -24,16 +24,32 @@ pub(crate) async fn collect_packages(
...
-            purls.insert(k.clone(), gr.purls[k].clone());
+    let mut errors = Vec::new();
+
+    // Merge the results into a total aggregate.
+
+    for collected in results {
+        for k in collected.purls.keys() {
+            // If there are already vulnerabilities recorded for a given
+            // package, we need to extend the existing list. If not,
+            // we initialize the list with the first bolus of results.
```
Comment (jcrossley3) [inline collectorist/api/src/server/collect.rs:40]: nice word!

**Review by jcrossley3 — APPROVED** [review]
The comments are helpful!

File: collector/osv/Cargo.toml (hunk 1 above)
Comment (bobmcwhirter) [reply collector/osv/Cargo.toml:31]: I assumed it's mostly "I don't care, and trust someone in the workspace who has a strong opinion to influence it"

**Pull request merged**